File: iotOnDjango/models/NetworkProfile.py

from django.db import models
from iotOnDjango.models.ConnectivityType import ConnectivityType

#======================================================================
# Class NetworkProfile Declaration
#======================================================================
class NetworkProfile (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	profileName = models.CharField(max_length=200, null=True)
	ssid = models.CharField(max_length=200, null=True)
	apn = models.CharField(max_length=200, null=True)
	device = models.ForeignKey('IoTDevice', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	gateway = models.ForeignKey('Gateway', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	simCard = models.ForeignKey('SimCard', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	connectivityType = models.CharField(max_length=64, null=True, choices=[(tag.name, tag.value) for tag in ConnectivityType])

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.profileName
		str = str + self.ssid
		str = str + self.apn
		str = str + self.connectivityType
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "NetworkProfile";
    
	def objectType(self):
		return "NetworkProfile";

File: iotonaspdotnet/Domain/NetworkProfile.cs
using iotonaspdotnet.Contracts;

namespace iotonaspdotnet.Domain;

public class NetworkProfile
{
    public Guid Id { get; set; } = Guid.NewGuid();

 public virtual long? NetworkprofileId { get; set; }
 public virtual string? ProfileName { get; set; }
 public virtual string? Ssid { get; set; }
 public virtual string? Apn { get; set; }
public virtual IoTDevice? Device { get; set; }
public virtual Gateway? Gateway { get; set; }
public virtual SimCard? SimCard { get; set; }
 public virtual ConnectivityType? ConnectivityType { get; set; }

    public static NetworkProfile FromRequest(NetworkProfileRequest request) {
        return new NetworkProfile {
            Id = request.Id,
            ProfileName = request.ProfileName,
            Ssid = request.Ssid,
            Apn = request.Apn,
            ConnectivityType = request.ConnectivityType,
        };
    }
}
